tests/lib/test_openstack_flavor.py

```diff
@@ -0,0 +1,140 @@
+            obj.name = i
+
+        mock_dest_names = [NonCallableMock() for _ in range(3)]
+        for i, obj in enumerate(mock_dest_names):
+            obj.name = i + 2
+
+        mock_side_effect = [mock_source_names, mock_dest_names]
+
+        self.instance.list_flavor = Mock(side_effect=mock_side_effect)
+        return_diff = self.instance.get_missing_flavors(
+            NonCallableMock(), NonCallableMock()
+        )
+        assert [0, 1] == return_diff
+
+    def test_create_flavor(self):
+        """
+        Test we can create a new flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor_data = NonCallableMock()
+
+        self.instance.create_flavor(mocked_cloud_name, mocked_flavor_data)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
+
+        self.api.create_flavor.assert_called_once_with(
+            name=mocked_flavor_data.name,
+            ram=mocked_flavor_data.ram,
+            vcpus=mocked_flavor_data.vcpus,
+            disk=mocked_flavor_data.disk,
+            flavorid="auto",
+            ephemeral=mocked_flavor_data.ephemeral,
+            swap=mocked_flavor_data.swap,
+            rxtx_factor=mocked_flavor_data.rxtx_factor,
+            is_public=mocked_flavor_data.is_public,
+        )
+
+    def test_empty_dict(self):
+        """
+        Test that method set_flavor_specs not called when empty_dictionary passed
+        """
+        empty_flavor_spec = {}
+        mocked_cloud_name = NonCallableMock()
+        flavor_id = NonCallableMock()
+        self.instance.set_flavor_specs(mocked_cloud_name, flavor_id, empty_flavor_spec)
+        self.api.set_flavor_specs.assert_not_called()
+
+    def test_get_specs(self):
+        """
+        Test getting the extra specs from a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        mocked_flavor = NonCallableMock()
+
+        self.instance.get_flavor(mocked_cloud_name, mocked_flavor)
+        self.instance.get_flavor_specs(mocked_cloud_name, mocked_flavor)
+
+    def test_set_specs(self):
+        """
+        Test setting the extra specs for a flavor
+        """
+        mocked_cloud_name = NonCallableMock()
+        flavor_id = NonCallableMock()
+        extra_specs = NonCallableMock()
+
+        self.instance.set_flavor_specs(mocked_cloud_name, flavor_id, extra_specs)
+        self.mocked_connection.assert_called_once_with(mocked_cloud_name)
+
+    def test_migrate_flavors(self):
+        """
+        Test that the migrate_flavors method behaves as expected
+        """
+        mocked_source_cloud = NonCallableMock()
+        mocked_dest_cloud = NonCallableMock()
+        source_flavor = Mock()
+        dest_flavor = self.instance.create_flavor(mocked_dest_cloud, source_flavor)
+        assert dest_flavor == self.api.create_flavor.return_value
+        self.instance.get_flavor_specs(mocked_source_cloud, dest_flavor.id)
+        self.instance.set_flavor_specs(
```

Comment:
Are these missing `.assert_called_once_with`? as it's making the method calls manually 

Also it doesn't look like `migrate_flavor` is actually called in this test?